### PR TITLE
Truncate attributes that exceed the available size

### PR DESCRIPTION
**Pull request by ThomasGooijers**
This applies the checks that are described in the documentation for `emAfReadOrWriteAttribute` before triggering the `emberAfExternalAttributeReadCallback` with the desired size.

The described checks where implemented already but only for attributes that are stored internally in `typeSensitiveMemCopy` 

This seems to have resolved our issue

### CODE REVIEW

File: src/app/util/attribute-storage.cpp
Comment (ThomasGooijers) [inline src/app/util/attribute-storage.cpp:644]: Alternatively we could pass readLength instead off emberAfAttributeSize(am) and implement these checks in our implementation for this callback. 
I left it here for now, what do you guys think?

File: src/app/util/attribute-storage.cpp
Comment (MauritsFassaert) [inline src/app/util/attribute-storage.cpp]: t_oo_ large. 

Also this is not an error. There's no ChipLogWarning, but maybe use `ChipLogDetail` ?

File: src/app/util/attribute-storage.cpp
Comment (MauritsFassaert) [reply src/app/util/attribute-storage.cpp:644]: I think then we may have a lot of built in clusters with external attributes to check.  They should not, but they may also not handle the `readLength < emberAfAttributeSize` case correctly and then there is new regression.

**Review by MauritsFassaert — APPROVED** [review]
approved with two small suggestions

File: src/app/util/attribute-storage.cpp
Comment (n-nievergeld) [inline src/app/util/attribute-storage.cpp]: This log message is not necessarily true. If the target only wants to put in a small string, it is not truncated. 
The question is: how should we reword:

Something like: "attribute size exceeds maximum read size"?

**Review by n-nievergeld — APPROVED** [review]
I approved: small comment about logmesssage.

Do you have a corresponding PR in main repo where the submodule is updated?

File: src/app/util/attribute-storage.cpp
Comment (ThomasGooijers) [reply src/app/util/attribute-storage.cpp]: I was looking for the warning indeed but ended up going with error since there wasn't one. 
But since we can handle it detail will do.

**Comment by ThomasGooijers** [comment]
> I approved: small comment about logmesssage.
> 
> Do you have a corresponding PR in main repo where the submodule is updated?

I was waiting until this was merged, I do now.

**Pull request merged**
